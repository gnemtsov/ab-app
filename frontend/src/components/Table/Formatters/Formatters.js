import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

// Formatter returns either:
// 1) string
// 2) object with JSX field and toString method

export const departmentLinker = (col, row) => {
		const jsx = <Link to={`/edit?d_id=${row.d_id}`}>
			{row.d_title}
		</Link>;
		return {
			JSX: jsx,
			toString: () => row.d_title.toString()
		}
	}

export const date = (col, row) => moment(row[col.name]).format('DD/MM/YYYY').toString();
