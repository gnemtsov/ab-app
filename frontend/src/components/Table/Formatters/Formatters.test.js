import React from 'react';

import { configure, render, mount, shallow } from 'enzyme';
import createRouterContext from 'react-router-test-context';
import Adapter from 'enzyme-adapter-react-16';

import * as Formatters from './Formatters';

configure({ adapter: new Adapter() });

describe('Formatters', () => {
	it('Render departmentLinker', () => {
		const f = Formatters.departmentLinker({}, {d_id: 1, d_title: 'test title'});
		expect(f.JSX).toBeDefined();
		expect(f.toString).toBeDefined();
		expect(f.toString()).toBe('test title');
		
		const context = createRouterContext();
		const wrapper = shallow (
	        f.JSX,
			{ context }
		);
		expect(wrapper).toMatchSnapshot();
	});
	
	it('Formats date', () => {
		expect(Formatters.date({name: 'date'}, {date: '20151026T2030'})).toBe('26/10/2015');
		expect(Formatters.date({name: 'date'}, {date: '2016-10-26'})).toBe('26/10/2016');
		expect(Formatters.date({name: 'date'}, {date: '20171026'})).toBe('26/10/2017');
		expect(Formatters.date({name: 'date'}, {date: '2018-10-26T15:27:00.000Z'})).toBe('26/10/2018');
	})
});
