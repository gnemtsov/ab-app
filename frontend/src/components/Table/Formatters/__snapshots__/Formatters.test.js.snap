// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formatters Render departmentLinker 1`] = `
<a
  href="/edit?d_id=1"
  onClick={[Function]}
>
  test title
</a>
`;
